### ISSUE TITLE
Angle-based Outlier Detector (ABOD) returns None

### ISSUE THREAD

**Issue by tankz0r** [yzhao062/Pyod#10]
Can you explain in which situations ABOD returns None and how should I interpret these?

**Comment by yzhao062** [comment]
Could you paste the code that cause the issue? Additionally, if the data is publicly available, could you share so I could reproduce the problem.

ABOD involves heavy distance/matrix calculation, which may be problematic. It may be a good idea to check:

1. Whether the data contains NaN, i.e. missing values
2. It could be useful to conduct normalization on features before using ABOD, e.g. using sklearn.StandardScaler()

Let me know if any of these works.
Best.

**Comment by tankz0r** [comment]
Sorry, but I cant share data. The code is taken from notebooks ```Benchmark.ipynb```. Before I put data into ABOD I fill missing values and use scaler as well. Nans occur only in ABOD, all others algorithms do not have any...

**Comment by yzhao062** [comment]
For returning "None" issue, I may provide more warning/error messages in the next release. I guess it is caused by some numerical overflow/underflow in calculating distance and the variance of weighted cosine similarity. Since I do not have the data so it is hard to know which part specifically causes the problem.

As for the parallelization, some models are enabled with this feature. Simply setting n_jobs=x should do the trick when initialization. They are:

1. KNN
2. IFOREST
3. LOF
4. MCD
5. OCSVM
6. PCA

Some are pending to add this feature: Feature_Bagging. 
Some are not planned to include this feature  for now: ABOD, CBLOF and HBOS.

From execution perspective, I would suggest IFOREST, HBOS and PCA for large datasets and prevent using  ABOD and Feature Bagging which could be time consuming.

I am also planning to add a benchmark table shortly to compare model accuracy and computational expenses across all implemented models.
